src/extensions/step_handlers/wiot_handler/src/wiot_handler_1.cpp

```diff
@@ -0,0 +1,353 @@
+    {
+        return Cancel(workflowData);
+    }
+
+    if (!IsValidUpdateTypeInfo(workflowHandle))
+    {
+        return ADUCResult(ADUC_Result_Failure, ADUC_ERC_UPDATE_CONTENT_HANDLER_INSTALL_FAILURE_UNKNOWNEXCEPTION);
+    }
+
+    const size_t fileCount = workflow_get_update_files_count(workflowHandle);
+    if (fileCount != 1)
+    {
+        // For v1, only 1 file is expected.
+        Log_Error("Incorrect file count: %u", fileCount);
+        return ADUCResult(ADUC_Result_Failure, ADUC_ERC_UPDATE_CONTENT_HANDLER_INSTALL_FAILURE_UNKNOWNEXCEPTION);
+    }
+
+    //
+    // Check to see if the content is already installed.
+    //
+
+    if (IsInstalled(workflowData).ResultCode == ADUC_Result_IsInstalled_Installed)
+    {
+        return ADUCResult(ADUC_Result_Install_Success);
+    }
+
+    workflow_string_ptr workFolder(workflow_get_workfolder(workflowHandle));
+    if (workFolder.get() == nullptr)
+    {
+        Log_Error("Unable to get work folder");
+
+        return ADUCResult(ADUC_Result_Failure, ADUC_ERC_UPDATE_CONTENT_HANDLER_INSTALL_FAILURE_UNKNOWNEXCEPTION);
+    }
+
+    Log_Info("Installing from %s", workFolder.get());
+
+    for (size_t fileIndex = 0; fileIndex < fileCount; fileIndex++)
+    {
+        if (workflow_is_cancel_requested(workflowHandle))
+        {
+            return Cancel(workflowData);
+        }
+
+        ADUC_FileEntity* entity_temp;
+        if (!workflow_get_update_file(workflowHandle, fileIndex, &entity_temp))
+        {
+            return ADUCResult(ADUC_Result_Failure, ADUC_ERC_UPDATE_CONTENT_HANDLER_INSTALL_FAILURE_UNKNOWNEXCEPTION);
+        }
+        workflow_file_entity_ptr entity(entity_temp);
+
+        std::string filePath(workFolder.get());
+        filePath += "/";
+        filePath += entity->TargetFilename;
+
+        Log_Info("Installing file #%u: %s", fileIndex, filePath.c_str());
+
+        if (!StepHandler::Install(filePath.c_str()))
+        {
+            Log_Error("Install failed");
+            return ADUCResult(ADUC_Result_Failure, ADUC_ERC_UPDATE_CONTENT_HANDLER_INSTALL_FAILURE_UNKNOWNEXCEPTION);
+        }
+    }
+
+    return ADUCResult(ADUC_Result_Install_Success);
+}
+
+/**
+ * @brief Apply implementation
+ *
+ * @return ADUC_Result The result of the apply.
+ * ADUC_Result_Success - success
+ * ADUC_Result_Apply_RequiredImmediateReboot - reboot required
+ */
+ADUC_Result WiotHandler1::Apply(const ADUC_WorkflowData* workflowData)
+{
+    Log_Info(HANDLER_LOG_ID "Apply");
+
+    const ADUC_WorkflowHandle workflowHandle{ workflowData->WorkflowHandle };
+
+    if (workflow_get_operation_cancel_requested(workflowHandle))
+    {
+        return Cancel(workflowData);
+    }
+
+    workflow_string_ptr workFolder(workflow_get_workfolder(workflowHandle));
+    if (workFolder.get() == nullptr)
+    {
+        Log_Error("Unable to get work folder");
+
+        return ADUCResult(ADUC_Result_Failure, ADUC_ERC_SCRIPT_HANDLER_APPLY_FAILURE_UNKNOWNEXCEPTION);
```

Comment:
Should probably make own ADUC_ERC group for wiot handler instead of overloading script handler one